public/app/features/explore/Graph/ExploreGraph.tsx

```diff
@@ -120,16 +120,18 @@ export function ExploreGraph({
   const dataLinkPostProcessor = useExploreDataLinkPostProcessor(splitOpenFn, timeRange);
 
   const dataWithConfig = useMemo(() => {
-    return applyFieldOverrides({
-      fieldConfig: styledFieldConfig,
-      data,
-      timeZone,
-      replaceVariables: (value) => value, // We don't need proper replace here as it is only used in getLinks and we use getFieldLinks
-      theme,
-      fieldConfigRegistry,
-      dataLinkPostProcessor,
-    });
-  }, [fieldConfigRegistry, data, timeZone, theme, styledFieldConfig, dataLinkPostProcessor]);
+    return loadingState === LoadingState.Done
+      ? applyFieldOverrides({
+          fieldConfig: styledFieldConfig,
+          data,
+          timeZone,
+          replaceVariables: (value) => value, // We don't need proper replace here as it is only used in getLinks and we use getFieldLinks
+          theme,
+          fieldConfigRegistry,
+          dataLinkPostProcessor,
+        })
+      : [];
```

Comment:
i guess returning an empty array here can cause an intermediate state with nothing rendered?

i think the purpose of the "Loading" state is just to make sure the panel chrome rendered the progress indicator. nothing else should change. ideally we would render the panel with the previous time range and previously processed frames?

maybe a couple `usePrevious()` hooks can do this?

seems possible, since we control all the inputs to PanelRenderer here.

![image](https://github.com/grafana/grafana/assets/43234/ea8f859c-5080-45b1-a53b-e23c134b33a3)

Reply:
Okay, I changed the time range to be the previous range if it exists and the loading state is loading. Is this what you have in mind?

Reply:
yep think so. will test when home :)